storage/src/main/java/com/brandwatch/ahmedanalytics/storage/StorageJob.java

```diff
@@ -0,0 +1,60 @@
+package com.brandwatch.ahmedanalytics.storage;
+
+import com.brandwatch.ahmedanalytics.common.entities.Mention;
+import com.brandwatch.ahmedanalytics.common.repositories.MentionsRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class StorageJob extends Thread{
```

Comment:
While this is probably ok in your case, it is generally prefered implement `Runnable` over extending `Thread`. 

A couple of resources, also about Executors:

https://www.baeldung.com/java-runnable-vs-extending-thread
https://javarevisited.blogspot.com/2016/12/difference-between-thread-and-executor.html
https://winterbe.com/posts/2015/04/07/java8-concurrency-tutorial-thread-executor-examples/

Reply:
I just override `run()` so there is no need for `Thread` as u said. I changed it to be runnable but I am not sure if this is the typical way to do it but it works fine.